### PR TITLE
Adds Anymal-C navigation rsl_rl example with obstacle and camera in the scene

### PR DESCRIPTION
**Pull request by KingsleyLiu-NV**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
List any dependencies that are required for this change.

This PR adds the anymal-c navigation rsl_rl example with obstacle and camera in the scene:

- Extends the rsl_rl libraries to support convolution networks and ppo runner
- Extends the anymal-c navigation environment to have rigid object obstacle and tiled camera sensor
  - Adds the event reset function to sample the position in a sector of ring area
  - Adds the observation function for the flattened normalized rgb data
  - Adds the pose3 command resample function to determine the target based on the obstacle position
- Tune the RL reward terms to make the training converge

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

It can be run with the following command:
```
./isaaclab.sh -p scripts/reinforcement_learning/rsl_rl/train.py --task Isaac-Navigation-Flat-Obstacle-Anymal-C-v0 --headless --num_envs 4096 --max_iterations 1501 --enable_cameras

./isaaclab.sh -p scripts/reinforcement_learning/rsl_rl/play.py --task Isaac-Navigation-Flat-Obstacle-Anymal-C-Play-v0 --headless --num_envs 16 --video
```

## Screenshots

The training results are as below:
image

Memory footprint: (num_envs=4096, image_input_shape=(3, 64, 64))
image

Training speed:
image

Play with `model_1300.pt`:

https://github.com/user-attachments/assets/b1eed6b5-f570-4eaf-9b62-27e1bcfbafb3

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
  - [x] isaaclab changelog & extension.toml
  - [x] isaaclab_rl changelog & extension.toml
  - [x] isaaclab_tasks changelog & extension.toml
  - [x] environments.rst
  - [x] ??? More
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/navigation/config/anymal_c/navigation_obstacle_env_cfg.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+    @configclass
+    class PolicyCfg(ObsGroup):
+        """Observations for policy group."""
+
+        # observation terms (order preserved)
+        base_lin_vel = ObsTerm(func=mdp.base_lin_vel)
+        projected_gravity = ObsTerm(func=mdp.projected_gravity)
+        pose_command = ObsTerm(func=mdp.generated_commands, params={"command_name": "pose_command"})
+        camera_data = ObsTerm(func=mdp.image_flatten, params={"sensor_cfg": SceneEntityCfg("tiled_camera")})
```
Comment (Toni-SM) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/navigation/config/anymal_c/navigation_obstacle_env_cfg.py]: This implementation flattens and concatenates all the observations components (propio and image) in one vector (a flattened `observation_space`), leading the definition of the image shape to use in the model to the agent configuration `image_input_shape`, right?

Agent configuration must be agnostic to any environment information except those specified by the environment spaces (`observation_space`, `state_space` and `action_space`). Currently, this is only possible with the direct workflow and using composite spaces

Comment (KingsleyLiu-NV) [reply source/isaaclab_tasks/isaaclab_tasks/manager_based/navigation/config/anymal_c/navigation_obstacle_env_cfg.py]: Yes, that understanding is correct. And I use a check to ensure the `image_input_shape` is consistent with the tiled camera configuration: [train.py#L132-L140](https://github.com/isaac-sim/IsaacLab/blob/fea/navigation-obstacle-camera/scripts/reinforcement_learning/rsl_rl/train.py#L132-L140)

File: scripts/reinforcement_learning/rsl_rl/actor_critic_conv2d.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+# SPDX-License-Identifier: BSD-3-Clause
+
+import torch
+import torch.nn as nn
+from torch.distributions import Normal
+
+from rsl_rl.modules.actor_critic import get_activation
+
+
+class ResidualBlock(nn.Module):
```
Comment (Mayankm96) [inline scripts/reinforcement_learning/rsl_rl/actor_critic_conv2d.py]: Since this is RSL-RL library related, I would recommend making an MR there for this actor critic structure :)

**Review by Mayankm96 — CHANGES_REQUESTED** [review]
A lot of the modifications in this MR should be handled by RSL-RL library and not IsaacLab. It would be good to keep the two separated to keep IsaacLab mainly as simulation infrastructure.

Would it be possible to make an MR to RSL-RL instead? We can review it there and deal with this usecase accordingly. It shouldn't be hard for us to support a dictionary of observations and avoid an ugly hack here to deal with it.

**Comment by KingsleyLiu-NV** [comment]
> A lot of the modifications in this MR should be handled by RSL-RL library and not IsaacLab. It would be good to keep the two separated to keep IsaacLab mainly as simulation infrastructure.
> 
> Would it be possible to make an MR to RSL-RL instead? We can review it there and deal with this usecase accordingly. It shouldn't be hard for us to support a dictionary of observations and avoid an ugly hack here to deal with it.

@Mayankm96 thanks for the suggestions. Basically, only `actor_critic_conv2d.py` and `on_policy_runner_conv2d.py` should go to RSL-RL library, and I can file an PR there.

Regarding `support a dictionary of observations and avoid an ugly hack here to deal with it`, can you elaborate what can be improved in this PR after conv2d-related code goes to RSL-RL library?

File: scripts/reinforcement_learning/rsl_rl/actor_critic_conv2d.py (hunk 2 above)
Comment (KingsleyLiu-NV) [reply scripts/reinforcement_learning/rsl_rl/actor_critic_conv2d.py]: Yes, I can do that. Then I suppose this PR can only be merged after the RSL-RL accepts my changes.

**Comment by Toni-SM** [comment]
> Regarding support a dictionary of observations and avoid an ugly hack here to deal with it, can you elaborate what can be improved in this PR after conv2d-related code goes to RSL-RL library?

As I commented previously, agent configuration must be agnostic to any environment information except those specified by the environment spaces (`observation_space`, `state_space` and `action_space`) and this is currently, this is only possible with the direct workflow and using [composite spaces](https://gymnasium.farama.org/api/spaces/composite/) (`Tuple` or `Dict`).

So, the current task implementation is a hacky and particular hard-code solution for one RL library that doesn't follow the convention to express highly structured space specification, and it is not generalizable/usable for other RL libraries. Since only Direct workflow support heterogeneous observations currently, the task implementation must be migrated to that workflow.

**Comment by Mayankm96** [comment]
I would push back on asking the user here to migrate their environment to direct workflow. That has its own set of inconveniences. We need to properly support heterogenous observation/action spaces to the manager-based environments. This is in the release plan for 2.2 of IsaacLab. We should probably prioritize it more as it is something doable.

@KingsleyLiu-NV An intermediate solution is that you have a separate observation group for the images similar to how we define the group called 'policy'. If this group is called 'images', you can access it from RSL-RL side through the dictionary: `extras["observations"]["images"]`. This is how we are dealing with asymmetric actor-critic over there. More information here: https://github.com/leggedrobotics/rsl_rl/blob/main/rsl_rl/runners/on_policy_runner.py#L32-L41

@kellyguo11 for visibility.

**Comment by Toni-SM** [comment]
> This is in the release plan for 2.2 of IsaacLab. We should probably prioritize it more as it is something doable.

For sure, adding to the Manager-based workflow the ability to work with different space specifications is something to be prioritized.

However, while that happens, we should indicate the appropriate workflows to create well-done tasks according to the agreed interface (that allow the project to be used by any RL library) and stop proposing/encouraging hard-code solution (that although it is not as *"ugly"* as the previous one, still hacky hard-code) tied to specific implementations that do not promote code reusability and compromise the robust, maintainable, and flexible codebase of the project. 

Therefore, for the time being, the way forward should be through Direct workflow by defining composite [observation_space](https://isaac-sim.github.io/IsaacLab/main/source/api/lab/isaaclab.envs.html#isaaclab.envs.DirectRLEnvCfg.observation_space).

**Comment by KingsleyLiu-NV** [comment]
Hi @Mayankm96 , I got your point of using `extras["observation"]`. However, I think there still requires a new branch in the exporter: https://github.com/isaac-sim/IsaacLab/blob/main/source/isaaclab_rl/isaaclab_rl/rsl_rl/exporter.py#L123-L151. If this kind of modification is acceptable, I will make the changes accordingly.

**Comment by explore-code-world7** [comment]
@ Kingsley Liu 

Hi, pro player, it's sorry to disturb you, I am a newbie.
Can I ask which measures did  you take to run scene with camera in headless mode?
I can't do it, i would appreciate it  if you could spare some time to check [my bug report](https://github.com/isaac-sim/IsaacLab/issues/2573)
Thank you a lot.

File: scripts/reinforcement_learning/rsl_rl/train.py
Comment (ooctipus) [inline scripts/reinforcement_learning/rsl_rl/train.py:1]: Thanks for creating the PR, how to edit code in train.py and play.py is probably going to wait until the rsl_rl code support for cnn to become available. in ideal case, cnn will be naturally supported without any modification to train.py and play.py needed XD

File: source/isaaclab/isaaclab/envs/mdp/commands/pose_2d_command.py
Comment (ooctipus) [inline source/isaaclab/isaaclab/envs/mdp/commands/pose_2d_command.py:1]: Why sample the goal position by using obstacles root_pos? what if 
`2 * self.object.data.root_pos_w[env_ids, 0] - self.robot.data.root_pos_w[env_ids, 0]` is within obstacle mesh? or if it is outside the terrain boundary if using non-Flat Terrain? e.g. Terrain Generator. If there are multiple objects does it also work? How do you guarantee the result goal position is not ill-form if terrains and objects is been modified? If the command term is not general enough, it is recommented to implemented in task-level mdp rather than in env.mdp. If you want to write is env.mdp, maybe we need better formulate the command sampling strategy.

**Comment by shahizat** [comment]
Hello all, has anyone managed to get observation navigation tasks working using depth and image observation with RSL-RL? From where can I see examples?
